onnx/defs/math/defs.cc

```diff
@@ -3415,6 +3415,72 @@ static T get_scalar_value_from_tensor(const ONNX_NAMESPACE::TensorProto* t) {
   }
 }
 
+// Shape inference for DFT is shared with IDFT
+void discrete_fourier_transform_shape_inference(ONNX_NAMESPACE::InferenceContext& ctx, bool is_onesided)
+{
+    propagateElemTypeFromInputToOutput(ctx, 0, 0);
+    if (!hasInputShape(ctx, 0))
+    {
+        // If no shape is available for the input, skip shape inference...
+        return;
+    }
+
+    // In general the output shape will match the input shape exactly
+    // So initialize the output shape with the input shape
+    auto& input_shape = getInputShape(ctx, 0);
+    ONNX_NAMESPACE::TensorShapeProto result_shape_proto = input_shape;
+
+    // Get the axis where the DFT will be performed.
+    auto axis = static_cast<int>(getAttribute(ctx, "axis", 1));
+    auto axis_idx = axis % input_shape.dim_size();
```

Comment:
This is a bit different from the usual handling (eg.: see: https://github.com/onnx/onnx/blob/dad79aca26a276a7b8175b9add0614b5ebaa6f89/onnx/defs/controlflow/old.cc#L156 ). Also, doesn't look like modulo will work as needed here for C++? (https://stackoverflow.com/questions/11630321/why-does-c-output-negative-numbers-when-using-modulo )

Reply:
This should have been
    auto axis_idx = (input_shape.dim_size() + axis % input_shape.dim_size()) % input_shape.dim_size();
this will allow for arbitrary index with wrapping.

Is this behavior expected?

Reply:
No. All other ops require axis to be in the range [-rank, rank-1] and reject an axis greater than rank. So, it is preferable that this op do the same. Thanks.

Reply:
Im assuming that this is a numpy convention?
Seems kind of arbitrary to restrict the range.